### DISCUSSION TITLE
lsp go to definition will automatically change cwd, how to exclude that?

### DISCUSSION THREAD

**Discussion by xzbdmw** [nvim-tree/nvim-tree.lua#2662] [General]
This will result in big file tree to open which makes gd showing buffers slow, I guess the sync root with cwd is synchronized so I have to wait tree to open then my buffer will open, I can't find a way to ignore those kinds of changing cwd, from lspconfig I also can't find where to disable the automatically changing cwd's behaviour, anyone can help me?

**Comment by alex-courtis** [comment]
Can you reproduce this issue with [Clean Room Replication](https://github.com/nvim-tree/nvim-tree.lua/wiki/Troubleshooting#clean-room-replication)? lua-langage-server is available at the bottom however will need to be uncommented.

If you cannot reproduce it in a clean room, I suggest:
* disable nvim-tree and see if the issue occurs
* you bisect your plugins:  disable your plugins until the issue stops occurring.

**Reply by xzbdmw** [reply]
diasble git intregration solve the problem:)

**Reply by alex-courtis** [reply]
That's not great; git integration is important and should not do that.

What are your setup options?

**Reply by xzbdmw** [reply]
```lua
{
    "nvim-tree/nvim-tree.lua",
    version = false,
    dependencies = {
        "nvim-tree/nvim-web-devicons",
    },
    lazy = false,
    config = function()
        require("nvim-tree").setup({
            git = {
                enable = false,
            },
            -- filesystem_watchers = {
            --     enable = false,
            --     debounce_delay = 0,
            --     ignore_dirs = {},
            -- },
            actions = {
                open_file = {
                    quit_on_open = false,
                },
                change_dir = {
                    global = true,
                },
            },
            disable_netrw = true,
            sync_root_with_cwd = true,
            respect_buf_cwd = true,
            update_focused_file = {
                enable = true,
                update_root = true,
                ignore_list = {},
            },
            on_attach = my_on_attach,
            view = {
                width = 25,
                preserve_window_proportions = true,
            },
            renderer = {
                special_files = {},
                group_empty = true,
                icons = {
                    web_devicons = {
                        file = { color = false, enable = false },
                        folder = {
                            color = false,
                        },
                    },
                    show = {
                        git = false,
                        modified = false,
                        bookmarks = false,
                    },
                    glyphs = {
                        default = "",
                        folder = {
                            arrow_closed = ">",
                            arrow_open = "󱞩",
                            default = "",
                            open = "",
                            empty = "",
                            empty_open = "",
                            symlink = "",
                            symlink_open = "",
                        },
                    },
                },
            },
        })
    end,
}
```
with git enabled, the first NvimTreeToggle command takes half second to load, even in small repos, and log file don't show where the delay comes from

**Reply by alex-courtis** [reply]
Unfortunately I'm not seeing either issue when using your setup with the nvim-tree repo.

Lazy implies that you have other plugins running... please replicate with [Clean Room Replication](https://github.com/nvim-tree/nvim-tree.lua/wiki/Troubleshooting#clean-room-replication) and raise two issues.

With both issues, please include a repo that we may use to replicate the problem.
